### PR TITLE
[2607-DEV-602] Calendar refactor 6: test coverage (date utils, ICS snapshot, calendar e2e)

### PR DESCRIPTION
**Pull request by teamenjoyvd**
## Summary
- Unit tests for `app/(dashboard)/calendar/utils.ts` + `lib/format.ts` Sofia-timezone helpers, including DST spring-forward/fall-back boundary cases.
- Extracted `buildEventDescription()` from `feed.ics/route.ts` into `lib/server/calendar.ts` (behavior-preserving) so the Phase 1c ICS description format can be snapshot-tested without mocking auth/DB.
- New `e2e/calendar.spec.ts`: unauthenticated smoke covering month load, month navigation, event popup, agenda view, and category filter — runs under `mobile-390`/`desktop`, not `authenticated`.

## Session State
**Status:** DONE
**Completed:**
- [x] `app/(dashboard)/calendar/utils.test.ts` — 15 tests, all passing
- [x] `lib/format.test.ts` extended — `calMonth`/`calDay`/`formatDateMediumEn`/`formatDateLongEn`
- [x] `lib/server/calendar.ts` — `buildEventDescription()` extracted
- [x] `lib/server/calendar.test.ts` — snapshot coverage of the Phase 1c format
- [x] `e2e/calendar.spec.ts` — unauthenticated smoke
**Next:** Confirm `mobile-390`/`desktop` e2e projects pass green on the Vercel Preview (`preview-smoke.yml`) — this repo's local sandbox has a reproducible pre-existing Node-fetch flake reaching Supabase from the Next dev server under browser-driven load (confirmed present on a clean `main` checkout too, unrelated to this diff); `curl` against the same route succeeds every time, so this is specific to this local sandbox's networking, not the app or this change.

## Test plan
- [x] `npx vitest run app/(dashboard)/calendar/utils.test.ts lib/format.test.ts lib/server/calendar.test.ts` — 3 files, 50 tests passed
- [x] `npx tsc --noEmit` — clean
- [ ] CI green + Vercel Preview READY (confirm `mobile-390` e2e project in `preview-smoke.yml` actually runs `e2e/calendar.spec.ts` and passes)

Closes #602

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Calendar feed descriptions now consistently include available event details such as location, meeting link, and category.
  * Added calendar formatting support for month, day, and long- and medium-date displays.

* **Tests**
  * Expanded automated coverage for calendar navigation, event dialogs, agenda filtering, time-zone handling, date formatting, and calendar feed descriptions.

### CODE REVIEW

File: e2e/calendar.spec.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      await page.keyboard.press('Escape')
+      await expect(dialog).toBeHidden()
+    }
+
+    await visible(page, page.getByRole('button', { name: 'Agenda' })).click()
+    await expect(grid).not.toBeVisible()
+
+    const filterButton = visible(page, page.getByRole('button', { name: 'In-person' }))
+    await filterButton.click()
+    await expect(filterButton).toHaveAttribute('aria-pressed', 'true')
```
[the remark this answers is not in the document]
Comment (teamenjoyvd) [reply e2e/calendar.spec.ts:54]: Skipped: `cal.inPerson` -> "In-person" is a real, distinct translation key from `cal.personal` -> "Personal" (see lib/i18n/domains/calendar.ts) — this locator targets the format filter row (In-person/Online/Hybrid), not the category row (N21/Personal). This PR's own CI run (390px smoke vs preview, mobile-390 project) already exercised e2e/calendar.spec.ts against the live Vercel Preview and passed using this exact 'In-person' locator, confirming it resolves correctly as written.
